### PR TITLE
feat: create a node info endpoint

### PR DESCRIPTION
**Pull request by darrenvechain**
# Description

Create a `/node/info` endpoint to retrieve details about the node

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- [ ] E2E Test
- [ ] Unit Test
- [x] Manual Test
```bash
curl http://localhost:8669/node/info | jq
```

Output:
```json
{
  "version": "2.1.3",
  "logsEnabled": true,
  "logsLimit": 1000,
  "debugEnabled": true,
  "debugCustomTracer": false,
  "debugPprofEnabled": false,
  "callGasLimit": 50000000,
  "wsBackTraceLimit": 1000
}
```

**Test Configuration**:

* Go Version:
* Hardware:
* Docker Version:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] New and existing E2E tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have not added any vulnerable dependencies to my code

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -57,6 +57,15 @@ func New(
 	}
 
 	router := mux.NewRouter()
+	info := node.Info{
+		Version:           doc.Version(),
+		LogsEnabled:       !skipLogs,
+		LogsLimit:         logsLimit,
+		DebugEnabled:      true,
```
[the remark this answers is not in the document]
Comment (darrenvechain) [reply api/api.go:64]: Its always enabled right now on master, but there is a PR open to have this flag

File: api/node/node.go [hunk 2]
```diff
@@ -30,11 +32,19 @@ func (n *Node) handleNetwork(w http.ResponseWriter, req *http.Request) error {
...
+}
+
 func (n *Node) Mount(root *mux.Router, pathPrefix string) {
 	sub := root.PathPrefix(pathPrefix).Subrouter()
 
 	sub.Path("/network/peers").
 		Methods(http.MethodGet).
 		Name("node_get_peers").
 		HandlerFunc(utils.WrapHandlerFunc(n.handleNetwork))
+	sub.Path("/info").
```
[the remark this answers is not in the document]
Comment (darrenvechain) [reply api/node/node.go:46]: Ya, this is a subpath, `/node` is configured above
```go
sub := root.PathPrefix(pathPrefix).Subrouter()
```

[the remark this answers is not in the document]
Comment (leszek-vechain) [reply api/node/node.go:46]: gotcha, thanks

**Comment by darrenvechain** [comment]
Closed as not needed

**Pull request closed without merging**